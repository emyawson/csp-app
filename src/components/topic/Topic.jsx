import React from 'react';
import PropTypes from 'prop-types';

// Skapa action list
import '@ingka/action-list/style.scss';
import '@ingka/focus/style.scss';
import '@ingka/switch/style.scss';
import '@ingka/svg-icon/style.scss';
import '@ingka/aspect-ratio-image/style.scss';
import '@ingka/button/style.scss';
import '@ingka/forms/style.scss';
import { ActionListItem } from '@ingka/action-list';

const Topic = ({
  label, caption, clickHandler,
}) => (
        <ActionListItem
          control='navigational'
          onClick={clickHandler}
          label={label}
          caption={caption}
        />
);

Topic.propTypes = {
  label: PropTypes.string,
  caption: PropTypes.string,
  clickHandler: PropTypes.func,
};

export default Topic;
