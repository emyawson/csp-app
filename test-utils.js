import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
