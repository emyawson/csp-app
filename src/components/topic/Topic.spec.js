import React from 'react';
// import { render, screen } from "@testing-library/react";
import Topic from './Topic.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<Topic />, combinedState);
};

test('renders <Topic/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
