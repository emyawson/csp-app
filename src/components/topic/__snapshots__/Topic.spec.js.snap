// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Topic/> 1`] = `
<div>
  <li
    class="action-list-item"
  >
    <button
      class="action-list-item__action"
    >
      <span
        class="action-list-item__label-container"
      >
        <span
          class="action-list-item__label"
        />
      </span>
      <span
        class="action-list-item__chevron"
      >
        <svg
          aria-hidden="true"
          class="svg-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            clip-rule="evenodd"
            d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </li>
</div>
`;
